pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentselector/OfflineSegmentSelector.java

```diff
@@ -30,10 +30,17 @@
  * Segment selector for offline table.
  */
 public class OfflineSegmentSelector implements SegmentSelector {
+  private final SegmentLineageBasedSegmentSelector _segmentLineageBasedSegmentSelector;
```

Comment:
This seems a bit confusing:
1. One segment selector class contains another segment segment selector.
2. Even though both are named SegmentSelector, only one of them implements the SegmentSelector interface.

I think SegmentLineageBasedSegmentSelector needs to be renamed, since it is not really implementing the interface.

Reply:
I adopted the `SegmentPreSelector` model.